oauthenticator/github.py

```diff
@@ -191,8 +194,27 @@ async def authenticate(self, handler, data=None):
         auth_state['access_token'] = access_token
         # store the whole user model in auth_state.github_user
         auth_state['github_user'] = resp_json
-        # A public email will return in the initial query (assuming default scope).
-        # Private will not.
+        # If a public email is not available, an extra API call has to be made
+        # to a /user/emails using the access token to retrieve emails. The
+        # scopes relevant for this are checked based on this documentation:
+        # https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps#normalized-scopes
+        if not auth_state['github_user']['email'] and (
+            'user' in scopes or 'user:email' in scopes
+        ):
+            req = HTTPRequest(
+                self.github_api + "/user/emails",
+                method="GET",
+                headers=_api_headers(access_token),
+                validate_cert=self.validate_server_cert,
+            )
+            try:
+                resp_json = await self.fetch(req, "fetching user emails")
+            except HTTPClientError:
+                pass
```

Comment:
Perhaps we should let this error properly, as we actually expect this to work reliably - right?